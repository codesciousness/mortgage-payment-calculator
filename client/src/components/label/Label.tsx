import React from 'react';
import InputLabel from '@mui/material/InputLabel';

type LabelProps = {
    name: string;
    focused?: boolean;
};

const Label = ({ name, focused }: LabelProps): JSX.Element => {
    const id = name.replaceAll(' ', '');

    return (
        <InputLabel
            id={id}
            focused={focused}
            sx={{
                textAlign: 'left',
                margin: '0.25rem 0'
            }}
        >
            {name}
        </InputLabel>
    );
};

export default Label;