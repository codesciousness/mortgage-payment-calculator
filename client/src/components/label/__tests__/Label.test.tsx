import React from 'react';
import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import Label from '../Label';

const name = 'Label Name';
const focused = true;

describe('<Label />', () => {
    test('Label component is defined', () => {
        expect(<Label name={name}/>).toBeDefined();
    });
    test('component matches the snapshot', () => {
        const component = create(<Label name={name} focused={focused}/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
    test('renders correctly when name prop is provided', () => {
        render(<Label name={name} focused={focused}/>);
        expect(screen.getByText('Label Name')).toBeInTheDocument();
    });
});