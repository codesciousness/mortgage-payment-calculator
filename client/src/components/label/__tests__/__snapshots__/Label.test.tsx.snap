// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> component matches the snapshot 1`] = `
<label
  className="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused css-22ecsy-MuiFormLabel-root-MuiInputLabel-root"
  id="LabelName"
>
  Label Name
</label>
`;
